lib/utils.js

```diff
@@ -198,172 +198,6 @@ function getMetaValue(entity, definition, key, groupStrategy='first', defaultVal
     return defaultValue;
 }
 
-/**
- * From: https://github.com/usolved/cie-rgb-converter/blob/master/cie_rgb_converter.js
- * Converts RGB color space to CIE color space
- * @param {Number} red
- * @param {Number} green
- * @param {Number} blue
- * @return {Array} Array that contains the CIE color values for x and y
- */
-function rgbToXY(red, green, blue) {
-    // Apply a gamma correction to the RGB values, which makes the color
-    // more vivid and more the like the color displayed on the screen of your device
-    const rgb = gammaCorrectRGB(red, green, blue);
```

Comment:
Originally `rgbToXY` converter had this gamma correction built-in.

I've dropped it from `ColorRGB.toXY()` because I believe conversions shouldn't have any side-effects. Currently I don't have clear idea on where to put it back.

I'm thinking that maybe user input should have 3-rd option in `extractColorFromValue()` output. Then we could handle RGB input like

```javascript
if (newColor.rgb !== null) {
    const rgb = newColor.rgb;
    const xy = rgb.toXY();
    const xyCorrected = rgb.gammaCorrected().toXY();
    // this is related to problem with storing corrected vs raw input
    // what should be stored in case of RGB input?
    newState.color_mode = 'xy';
    newState.color = xy.toObject();  // TODO: conversion 0..1->0..255 is missing
    ...
}
```

I

Reply:
I think that is OK, I think this build in might also explain the slight shifts I see when doing hsv -> xy -> hsv again, which would make sense if it was applying corrections in the rgb step.

Reply:
> That would also allow us to do rgb -> hs conversion properly if we have bubs that only support hs... not that I know of any atm.

Tuya light source supports only HSV AFAIK

Reply:
So yeah, definitely a good idea then :)

Reply:
Ok, done

Reply:
If I understand correctly, this has now been added back here: https://github.com/Koenkk/zigbee-herdsman-converters/blob/03e0884d32a0e0dc568f474182588cdadd46f1b2/converters/toZigbee.js#L864  ?

Reply:
you're correct

first apply gamma correction, then do transformation, but in two separate steps